### PR TITLE
Fix editingwhileofferorcontract

### PR DESCRIPTION
**Pull request by Grimaldus12**
Fixing a few issues: 
  - items can now only be edited when not having any open contracts or pending offers
  - edit item cascade type excludes merge (necessary due to inheritance)

### CODE REVIEW

File: src/main/java/de/hhu/propra/sharingplatform/service/ItemService.java [hunk 1]
```diff
@@ -64,6 +71,7 @@ public Item findItem(long itemId) {
 
     public void editItem(Item newItem, long oldItemId, long userId) {
         Item oldItemRental = findItem(oldItemId);
+        ItemValidator.validateItemIsFree(offerRepo, contractRepo, oldItemRental);
```
Comment (DirtyDan007) [inline src/main/java/de/hhu/propra/sharingplatform/service/ItemService.java]: Validation should also take place on trying to open the edit view and not only on storing changes

Comment (Grimaldus12) [reply src/main/java/de/hhu/propra/sharingplatform/service/ItemService.java]: fixed

**Review by ik8472 — APPROVED** [review]

File: src/main/java/de/hhu/propra/sharingplatform/service/ItemService.java [hunk 2]
```diff
@@ -62,6 +69,11 @@ public Item findItem(long itemId) {
         return item;
     }
 
+    public void itemIsFree(long itemId) {
```
Comment (DirtyDan007) [inline src/main/java/de/hhu/propra/sharingplatform/service/ItemService.java:72]: very good

**Review by DirtyDan007 — APPROVED** [review]

**Comment by Grimaldus12** [comment]
Closed: See #237 for changes

**Pull request closed without merging**